api_app/serializers.py

```diff
@@ -179,10 +194,48 @@ def validate(self, attrs: dict) -> dict:
         file_obj.seek(0)
         file_buffer = file_obj.read()
         attrs["md5"] = calculate_md5(file_buffer)
+        self.filter_analyzers_and_connectors(attrs)
         logger.debug(f"after attrs: {attrs}")
         return attrs
 
 
+class MultipleObservableAnalysisSerializer(rfs.ListSerializer):
+    """
+    ``Job`` model's serializer for Multiple Observable Analysis.
+    Used for ``create()``.
+    """
+
+    def update(self, instance, validated_data):
+        raise NotImplementedError("This serializer does not support update().")
+
+    child = None
+    many = True
+    observables = rfs.ListField(required=True)
+
+    def to_internal_value(self, data):
+        ret = []
+        errors = []
+
+        common_data = dict(data)
+        common_data.pop("observables", None)
+        for classification, name in data.get("observables"):
+            item = copy.deepcopy(common_data)
+            item["observable_name"] = name
+            item["observable_classification"] = classification
+            try:
+                validated = self.child.run_validation(item)
```

Comment:
I didn't understand why you chose to use `deepcopy` here. Couldn't you just create a new dict and populate it with with the values?

Reply:
This `deepcopy` ensures that this code snippet doesn't break even if new fields are added in future

Reply:
could you add it as a comment? thanks for the clarification!

Reply:
Alright